rllib/evaluation/episode_v2.py

```diff
@@ -0,0 +1,338 @@
+import random
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Tuple
+
+from ray.rllib.env.base_env import _DUMMY_AGENT_ID
+from ray.rllib.evaluation.collectors.simple_list_collector import (
+    _AgentCollector,
+    _PolicyCollector,
+    _PolicyCollectorGroup,
+)
+from ray.rllib.policy.policy_map import PolicyMap
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.annotations import DeveloperAPI
+from ray.rllib.utils.typing import AgentID, EnvID, PolicyID, TensorType
+
+if TYPE_CHECKING:
+    from ray.rllib.agents.callbacks import DefaultCallbacks
+    from ray.rllib.evaluation.rollout_worker import RolloutWorker
+
+
+@DeveloperAPI
+class EpisodeV2:
+    """Tracks the current state of a (possibly multi-agent) episode."""
+
+    def __init__(
+        self,
+        env_id: EnvID,
+        policies: PolicyMap,
+        policy_mapping_fn: Callable[[AgentID, "EpisodeV2", "RolloutWorker"], PolicyID],
+        *,
+        worker: Optional["RolloutWorker"] = None,
+        callbacks: Optional["DefaultCallbacks"] = None,
+    ):
+        """Initializes an Episode instance.
+
+        Args:
+            env_id: The environment's ID in which this episode runs.
+            policies: The PolicyMap object (mapping PolicyIDs to Policy
+                objects) to use for determining, which policy is used for
+                which agent.
+            policy_mapping_fn: The mapping function mapping AgentIDs to
+                PolicyIDs.
+            worker: The RolloutWorker instance, in which this episode runs.
+        """
+        # Unique id identifying this trajectory.
+        self.episode_id: int = random.randrange(2e9)
```

Comment:
Is it possible to have truly unique episode ids? Random sampling like this doesn't make it unique, in fact I think there are some other places we do this which causes crashes/issues for long-running jobs